src/Header/Cookie.php

```diff
@@ -97,13 +97,30 @@ public function getFieldValue()
     {
         $nvPairs = [];
 
-        foreach ($this as $name => $value) {
+        $cookies = [];
+        foreach ($this->flattenCookies($cookies, $this) as $name => $value) {
             $nvPairs[] = $name . '=' . (($this->encodeValue) ? urlencode($value) : $value);
         }
 
         return implode('; ', $nvPairs);
     }
 
+    protected function flattenCookies(&$result, $data, $prefix = null)
```

Comment:
Why we need a recursive function here? Do we need to manage array of array values in cookies? I think we should simplify the implementation here, for instance just iterating on keys and values of $value, if it is an array.